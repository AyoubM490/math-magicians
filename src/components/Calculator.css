.calc-container {
  width: 400px;
  margin: 3rem auto;
}

.button-wrap {
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: repeat(4, 1fr);
  grid-template-columns: repeat(4, 1fr);
  background-color: #eee;
  align-items: center;
  justify-content: center;
}

.button,
.result {
  border: 1px solid #ddd;
  text-align: center;
  font-size: 16px;
  color: #333;
  padding: 1.5rem;
}

.button {
  cursor: pointer;
}

.span_5 {
  background-color: #888;
  -ms-grid-column: 1 / 5;
  grid-column: 1 / 5;
  text-align: right;
  padding: 1.5rem;
  color: #f1f1f1;
}

.button_3 {
  -ms-grid-column: 1 / 3;
  grid-column: 1 / 3;
}

.colored {
  background-color: rgb(245, 131, 1);
}
