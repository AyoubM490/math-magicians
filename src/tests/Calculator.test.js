import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/calculator';

describe('Test if Calculator', () => {
  test('renders Calculator correctly', () => {
    const component = renderer.create(<Calculator />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Calculator', () => {
  test('Responds on user pressing two numbers ', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('2'));
    expect(screen.getByRole('textbox').value).toBe('12');
  });
  test('Responds on user pressing one number ', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    expect(screen.getByRole('textbox').value).toBe('1');
  });
});
