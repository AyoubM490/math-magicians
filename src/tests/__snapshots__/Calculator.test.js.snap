// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Calculator renders Calculator correctly 1`] = `
<div
  className="calcWrapper"
>
  <div>
    <h2
      className="mdHeading"
    >
      Let's do some Math!
    </h2>
  </div>
  <div
    className="calc-container"
  >
    <div
      className="calc-wrap"
    >
      <div
        className="button-wrap"
      >
        <input
          className="button span_5 input"
          placeholder="0"
        />
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button colored"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button colored"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button colored"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button colored"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="button span_3"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button colored"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
